From 8ce72b602046e39c862c7db7eb885ca4b3ac960c Mon Sep 17 00:00:00 2001
From: Hugh Waite <hugh.waite@acision.com>
Date: Tue, 12 Aug 2014 19:56:45 +0100
Subject: [PATCH] core/parser: Free rr correctly in error cases

- Fix potential memory leak if parameters are parsed into rr structure
(cherry picked from commit b40d92962b680ac29dd38ffe3c8f1aa1e39d3a97)
---
 parser/parse_rr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parser/parse_rr.c b/parser/parse_rr.c
index d3a2938..e3df4f3 100644
--- a/parser/parse_rr.c
+++ b/parser/parse_rr.c
@@ -138,7 +138,7 @@ static inline int do_parse_rr_body(char *buf, int len, rr_t **head)
 	}
 
  error:
-	if (r) pkg_free(r);
+	if (r) free_rr(r);
 	free_rr(head); /* Free any contacts created so far */
 	return -1;
 
-- 
2.1.0.rc1

